tools/mtmd/models/lfm2-audio-enc.cpp

```diff
@@ -0,0 +1,267 @@
+            auto * scores = ggml_add(ctx0, matrix_ac, matrix_bd);
+            scores = ggml_scale(ctx0, scores, 1.0f / std::sqrt(d_head));
+            cb(scores, "conformer.layers.{}.self_attn.id0", il);
+
+
+            ggml_tensor * attn = ggml_soft_max(ctx0, scores);
+            // TODO(tarek): combine permutes
+            Vcur = ggml_cont(ctx0, ggml_permute(ctx0, Vcur, 0, 2, 1, 3));
+            Vcur = ggml_cont(ctx0, ggml_permute(ctx0, Vcur, 1, 0, 2, 3));
+            ggml_tensor * x = ggml_mul_mat(ctx0, attn, Vcur);
+            // TODO(tarek): combine permutes
+            x = ggml_cont(ctx0, ggml_permute(ctx0, x, 1, 0, 2, 3));
+            x = ggml_cont(ctx0, ggml_permute(ctx0, x, 0, 2, 1, 3));
+            x = ggml_reshape_2d(ctx0, x, x->ne[0] * x->ne[1], x->ne[2]);
+
+            x = ggml_mul_mat(ctx0, layer.o_w, x);
+            ggml_tensor * out = ggml_add(ctx0, x, layer.o_b);
+            cb(out, "conformer.layers.{}.self_attn.linear_out", il);
+
+            cur = out;
+        }
+
+        residual = ggml_add(ctx0, residual, cur);
+        cur = build_norm(residual, layer.norm_conv_w, layer.norm_conv_b, NORM_TYPE_NORMAL, 1e-5, il);
+        cb(cur, "conformer.layers.{}.norm_conv", il);
+
+        // conv
+        {
+            auto * x = cur;
+            auto * conv_pw1_w = ggml_reshape_2d(ctx0, layer.conv_pw1_w, layer.conv_pw1_w->ne[1], layer.conv_pw1_w->ne[2]);
+            x = ggml_mul_mat(ctx0, conv_pw1_w, x);
+            x = ggml_add(ctx0, x, layer.conv_pw1_b);
+            x = ggml_cont(ctx0, ggml_transpose(ctx0, x));
+            cb(x, "conformer.layers.{}.conv.pointwise_conv1", il);
+
+            x = ggml_cont(ctx0, ggml_transpose(ctx0, x));
+
+            // TODO: add support of torch.funtional.glu
+            {
+                int64_t d   = x->ne[0] / 2;
+                ggml_tensor *gate = ggml_sigmoid(ctx0, ggml_view_2d(ctx0, x, d, x->ne[1], x->nb[1], d * x->nb[0]));
+                x = ggml_mul(ctx0, ggml_view_2d(ctx0, x, d, x->ne[1], x->nb[1], 0), gate);
+                x = ggml_cont(ctx0, ggml_transpose(ctx0, x));
+            }
+
+            // use ggml_ssm_conv for f32 precision
+            x = ggml_pad(ctx0, x, 4, 0, 0, 0);
+            x = ggml_roll(ctx0, x, 4, 0, 0, 0);
+            x = ggml_pad(ctx0, x, 4, 0, 0, 0);
+            x = ggml_cont(ctx0, x);
+            auto * conv_dw_w = ggml_reshape_2d(ctx0, layer.conv_dw_w, layer.conv_dw_w->ne[0], layer.conv_dw_w->ne[2]);
+            x = ggml_ssm_conv(ctx0, x, conv_dw_w);
+            x = ggml_add(ctx0, x, ggml_reshape_1d(ctx0, layer.conv_dw_b, layer.conv_dw_b->ne[0]));
+            x = ggml_cont(ctx0, ggml_transpose(ctx0, x));
+
+            cb(x, "conformer.layers.{}.conv.depthwise_conv", il);
+
+            {
+                x = ggml_cont(ctx0, ggml_transpose(ctx0, x));
+                x = ggml_add(ctx0, ggml_mul(ctx0, x, layer.conv_norm_w), layer.conv_norm_b);
+                x = ggml_cont(ctx0, ggml_transpose(ctx0, x));
```

Comment:
we may be able to remove of transposes if `conv_norm_b` is already transpose upon conversion?